iroh-gateway/src/rpc.rs

```diff
@@ -1,16 +1,29 @@
 use std::result;
 
 use anyhow::Result;
+use futures::stream::{BoxStream, StreamExt};
 use iroh_rpc_client::{create_server, GatewayServer, ServerError, ServerSocket};
 use iroh_rpc_types::gateway::{
-    GatewayAddr, GatewayRequest, GatewayService, VersionRequest, VersionResponse,
+    GatewayAddr, GatewayRequest, GatewayService, VersionRequest, VersionResponse, WatchRequest,
+    WatchResponse,
 };
 use tracing::info;
 
 #[derive(Default, Debug, Clone)]
 pub struct Gateway {}
 
 impl Gateway {
+    #[tracing::instrument(skip(self))]
+    fn watch(self, _: WatchRequest) -> BoxStream<'static, WatchResponse> {
+        async_stream::stream! {
+            loop {
+                yield WatchResponse { version: env!("CARGO_PKG_VERSION").to_string() };
```

Comment:
Can we DRY the computation of the version? Make a fn version_string() or something?

Reply:
also should be a `OnceCell` or sth like that, this only needs to be pulled once per runtime

Reply:
any objection to making `VERSION` a constant?

`const VERSION: &str = env!(CARGO_PKG_VERSION);`

Gets set once at compile time.

Reply:
I belive that should work 👍